src/Database/Query.php

```diff
@@ -1941,7 +1941,7 @@ public function func(): FunctionsBuilder
      * @psalm-suppress ImplementedReturnTypeMismatch
      */
     #[ReturnTypeWillChange]
-    public function getIterator()
+    public function getIterator(): StatementInterface
```

Comment:
This is not `Traversable`. This needs to be resolved before return-types are non-tentative. 
https://www.php.net/manual/en/class.iteratoraggregate

Reply:
The fact that `StatementInterface` isn't traversable can be fixed by making it extend `Traversable`. But the larger issues is that `Database/Query::getIterator()` returns a `StatementInterface` instance while `ORM\Query::getIterator()` returns a `ResultSetInterface`. This is not reconcilable.